### DISCUSSION TITLE
Why is pyscript not the default home assistant python script integration?

### DISCUSSION THREAD

**Discussion by Risbo6** [custom-components/pyscript#450] [General]
I'm sure a lot of people start with the [default integration](https://www.home-assistant.io/integrations/python_script/) and eventually switch to pyscript 👍

**Comment by craigbarratt** [comment]
I originally submitted `pyscript` as a [PR to HASS core](https://github.com/home-assistant/core/pull/38017), but it was rejected.  Here's what @balloob said:

> Hey @craigbarratt, thanks for your PR. This is not a feature that we would want to add to the main Home Assistant repository at this point. Using custom integrations for automations or our existing Python Script integration never really got any traction and I doubt that this would go any farther, so wouldn't want to take over the maintenance burden by adding it to core.

Obviously at the time it was a relatively large and complex integration from an unknown author, and @balloob's explanation seemed very reasonable to me.  HACS provided a good path to make it available to the community.  @balloob was quite supportive in providing feedback and mentioning `pyscript` to the community after it became available via HACS.

It's unfortunate that the default integration requires a high level of expertise to be useful, and doesn't make it easy to write compact automations.  So it's no surprise that HASS hasn't had much adoption of python as a user scripting language, which is exactly what `pyscript` was designed to overcome.

If there was enough user support, perhaps @balloob could reconsider at some point?

**Comment by redlefloh** [comment]
I can't support this enough. Pyscript has been a game changer for me. It is amazingly powerful yet super intuitive, simple and reliable. The fact that I can just write a python script with direct access to the HA internals, put it in the pyscript directory and it runs is fantastic! And the combination of not needing to restart anything to just add new functionality as well as the ability to test and code with Jupyter notebooks is pure joy and actually even more powerful than the already super useful YAML template editor.  So thanks so much @craigbarratt for what you have done and of course huge, huge kudos to @balloob for everything you've done with HA and if the two would natively marry, it would just be a dream wedding I am sure for many!

**Reply by dy-nz** [reply]
Ditto from me.
In addition its limited "import" capability permits it to address some areas of weakness in HASS, for example, telnet for multi-zone amplifiers, pre-amplifiers, HDMI switches. The HASS switch and sensor options are not practical for devices with complex telnet command sets (and no other options like http).

**Reply by dy-nz** [reply]
PS, @balloob a huge shout out from me. HASS has been a fantastic replacement for my iRule environment - a lot more than I was expecting. And after bending my brain into the HASS way of doing things, it has been a lifesaver for me.

**Reply by j-steve** [reply]
Same, I honestly can't understand anyone not wanting to make PyScript a core part of HomeAssistant. Things in Home Assistant are so complex that you basically need to be a software engineer to get everything working, and if you _are_ a software engineer you probably writing code in a real language like Python.

**Reply by GollyJer** [reply]
Putting this here as a selfish little ping for help on this question since you guys seem super happy with your setups.
[How are you editing/creating files & folders?](https://github.com/custom-components/pyscript/discussions/507)
